src/hooks/useCombobox/index.js

```diff
@@ -170,309 +174,339 @@ function useCombobox(userProps = {}) {
     },
   )
   /* Event handler functions */
-  const inputKeyDownHandlers = {
-    ArrowDown(event) {
-      event.preventDefault()
-      dispatch({
-        type: stateChangeTypes.InputKeyDownArrowDown,
-        shiftKey: event.shiftKey,
-        getItemNodeFromIndex,
-      })
-    },
-    ArrowUp(event) {
-      event.preventDefault()
-      dispatch({
-        type: stateChangeTypes.InputKeyDownArrowUp,
-        shiftKey: event.shiftKey,
-        getItemNodeFromIndex,
-      })
-    },
-    Home(event) {
-      event.preventDefault()
-      dispatch({
-        type: stateChangeTypes.InputKeyDownHome,
-        getItemNodeFromIndex,
-      })
-    },
-    End(event) {
-      event.preventDefault()
-      dispatch({
-        type: stateChangeTypes.InputKeyDownEnd,
-        getItemNodeFromIndex,
-      })
-    },
-    Escape() {
-      dispatch({
-        type: stateChangeTypes.InputKeyDownEscape,
-      })
-    },
-    Enter(event) {
-      // if IME composing, wait for next Enter keydown event.
-      if (event.which === 229) {
-        return
-      }
-
-      if (isOpen && highlightedIndex > -1) {
+  const inputKeyDownHandlers = useMemo(
+    () => ({
+      ArrowDown(event) {
         event.preventDefault()
         dispatch({
-          type: stateChangeTypes.InputKeyDownEnter,
+          type: stateChangeTypes.InputKeyDownArrowDown,
+          shiftKey: event.shiftKey,
           getItemNodeFromIndex,
         })
-      }
-    },
-  }
-
-  // Event handlers.
```

Comment:
All of the event handlers have been moved inside the prop getters that used them. This made it much easier to memoize. Otherwise we'd have to memoize each of these as well (which is fine, but just extra work for both us and the JavaScript engine).